src/operator/api/v1alpha3/postgresqlserverconfig_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DBPermissionChange string
+
+const (
+	DBPermissionChangeApply  DBPermissionChange = "APPLY"
+	DBPermissionChangeDelete DBPermissionChange = "DELETE"
+)
+
+type DatabaseCredentials struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
```

Comment:
Does this mean we store the password plaintext on the CRD? Seems like bad practice.

Reply:
Going to be changed to a different CRD that holds credentials (and should have similar access permissions)